### ISSUE TITLE
Declarative cell editor

### ISSUE THREAD

**Issue by warpech** [handsontable/hot-table#5]
Dependent on #4

Should be possible to use a `<template rel="hot-editor">` as a cell editor, so one can declare editors with HTML. Any HTML form, Custom Element or Polymer variable can be used as a cell editor.

```html
<hot-column>
  <template rel="hot-editor">
    <template bind="{{ProductSearch}}">
      <input type="search" value="{{Query}}">
      <template repeat="{{Results}}">
        <span onclick="this.templateInstance.model.Choose$++">{{Name}}</span>
      </template>
    </template>
  </template>
</hot-column>
```

The editor MAY actually NOT make any direct change in Handsontable datasource. Other than that, it changes something in controller/server, and controller/server decides to update that datasource which will be reflected in Handsontable. Possible use scenario: Dropdown menu shows just how many people are invited to a event. Expanding the dropdown menu brings a searchable Handsontable instance with search and multiselect capabilities.

**Comment by tntclaus** [comment]
Hi, I tested editor functionality an it seems to work fine except a bug with polymer custom elements click processing (try to add to your custom element any polymer element like core-item, etc.). Your editor component is just getting closed when you click on it without triggering any on-click handlers. 
I can open an issue if you'll need additional data.

Besides that I want to ask you to add this line:
```javascript
element.templateInstance.editor = this;
```
to the end of this method:
```javascript
  /**
   * Create editor element from template
   */
  Editor.prototype.createEditorElement = function() {
    var element;

    if (this.container.firstChild) {
      return;
    }
    element = this.template.createInstance({});
    this.model = element.templateInstance.model;
    this.container.appendChild(element);
    element.templateInstance.editor = this;
  };
```
at editor.js. I think this edit is too small to ask for pull request. 

I didn't find any way to close editor right after data pick except calling ```finishEditing();``` but in my custom element I have no reference to the editor instance. If there is any other correct ways to do that I would appreciate to know more about them.

**Comment by budnix** [comment]
Thanks @tntclaus for raport. Functionality didn't get merged to develop yet so I will try to fix bug in this branch (feature/issue-5)

**Comment by budnix** [comment]
Released in v0.1.7
